### PR TITLE
[BACK-1341] blob api device logs - implementation 

### PR DESCRIPTION
**Pull request by jh-bate**
see #589 

- this PR is based off main and includes the missing tests that were stubbed out in the previous PR 
- NOTE: also split out the `blob/store/structured/mongo/*` files into `blob` and `deviceLogs` as it was getting large and confusing

### CODE REVIEW

**Comment by tjotala** [comment]
This should be reviewed by @toddkazakov or his delegate

File: blob/store/structured/mongo/blob_repository.go [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+			"$exists": false,
+		},
+	}
+	if filter.MediaType != nil {
+		query["mediaType"] = bson.M{
+			"$in": *filter.MediaType,
+		}
+	}
+	opts := storeStructuredMongo.FindWithPagination(pagination).
+		SetSort(bson.M{"createdTime": -1})
```
Comment (lostlevels) [inline blob/store/structured/mongo/blob_repository.go:101]: Should there be an index created on `createdTime`?

File: blob/store/structured/mongo/blob_repository.go [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+		UserID:      pointer.FromString(userID),
+		MediaType:   create.MediaType,
+		Status:      pointer.FromString(blob.StatusCreated),
+		CreatedTime: pointer.FromTime(now.Truncate(time.Millisecond)),
+		Revision:    pointer.FromInt(0),
+	}
+
+	var id string
+	var err error
+	for retry := 0; retry < 3; retry++ {
```
[the remark this answers is not in the document]
Comment (jh-bate) [reply blob/store/structured/mongo/blob_repository.go:145]: Good question - tbh I don't know as this the code was written by someone else and I just moved to a different file. I will have a dig and see what I can find

File: blob/store/structured/mongo/blob_repository.go [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+	return result, nil
+}
+
+func (b *BlobRepository) DeleteAll(ctx context.Context, userID string) (bool, error) {
+	ctx, logger := log.ContextAndLoggerWithField(ctx, "userId", userID)
+
+	if ctx == nil {
+		return false, errors.New("context is missing")
+	}
+	if userID == "" {
```
[the remark this answers is not in the document]
Comment (jh-bate) [reply blob/store/structured/mongo/blob_repository.go:176]: yeah this pattern is repeated in a number of places, including `blob.IsValidID` - I will update

File: blob/store/structured/mongo/blob_repository.go (hunk 1 above)
Comment (jh-bate) [reply blob/store/structured/mongo/blob_repository.go:101]: unsure - this was existing code

File: blob/service/api/v1/v1.go [hunk 4]
```diff
@@ -122,6 +126,73 @@ func (r *Router) Create(res rest.ResponseWriter, req *rest.Request) {
...
+		return
+	}
+	mediaType, err := request.ParseMediaTypeHeader(req.Header, "Content-Type")
+	if err != nil {
+		responder.Error(http.StatusBadRequest, err)
+		return
+	} else if mediaType == nil {
+		responder.Error(http.StatusBadRequest, request.ErrorHeaderMissing("Content-Type"))
+		return
+	} else if *mediaType != "application/json; charset=utf-8" {
```
[the remark this answers is not in the document]
Comment (jh-bate) [reply blob/service/api/v1/v1.go:150]: change check so that it just requires `application/json`

**Review by lostlevels — DISMISSED** [review]

**Review by lostlevels — APPROVED** [review]

**Pull request merged**